core/src/main/scala/org/http4s/internal/parsing/Rfc2234.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.internal.parsing
+
+import cats.parse.Parser1
+import cats.parse.Parser.{charIn}
+
+/** Common rules defined in RFC2234.
+  *
+  * @see [[https://tools.ietf.org/html/rfc2234] RFC2234, Augmented BNF for Syntax Specifications: ABNF]
+  */
+private[http4s] object Rfc2234 {
```

Comment:
Maybe this is a bit much: Rfc3986 points to Rfc2234 for things that are already defined a couple other places.  Alpha is alpha and digit is digit.  But I kind of like it for traceability.